File: algorithms/13_square_root.py
SMALLEST = 0.00001

number = float(input("Digite um numero: "))
b = 2

while abs(number - (b * b)) > SMALLEST:
    p = (b + (number / b)) / 2
    b = p

print(f"O resultado aproximado da raiz quadrada de {number:.2f} é {p:.2f}")

File: algorithms/13_square_root.c
#define SMALLEST 0.00001

#include <stdio.h>
#include <stdlib.h>

int main()
{
    double number;
    double b = 2;
    double p;

    printf("Digite um numero: \n");
    scanf("%lf", &number);

    while (abs(number - (b * b)) > SMALLEST)
    {
        p = (b + (number / b)) / 2;
        b = p;
    }

    printf("O resultado aproximado da raiz quadrada de %.2lf é %.2lf", number, p);

    return 0;
}